### PR TITLE
Fix memory leak in tracetools::get_symbol()

### PR DESCRIPTION
**Pull request by christophebedard**
Part of https://github.com/ros2/ros2_tracing/issues/34

Requires https://github.com/ros2/ros2_tracing/pull/43

Requires https://github.com/ros2/ros2_tracing/pull/46 to help avoid memory allocations when the tracepoint isn't enabled.

Signed-off-by: Christophe Bedard <christophe.bedard@apex.ai>

### CODE REVIEW

**Comment by christophebedard** [comment]
This is the simplest fix, but it could be improved. For example, we could first check if the tracepoint is actually enabled before calling `tracetools::get_symbol()` and allocating. This would require changes in `tracetools`, though. See https://github.com/ros2/ros2_tracing/pull/43#issuecomment-1429001945

**Review by alsora — APPROVED** [review]

**Comment by alsora** [comment]
Isn't the macro sufficient to prevent using this code if tracetools is disabled?
Or do we support both "compile time disabled" and "runtime disabled" ?

**Comment by mjcarroll** [comment]
> Or do we support both "compile time disabled" and "runtime disabled" ?

When building with tracepoints enabled, there will be some runtime associated here, even if the tracepoints are not actively being used.  I think this is more important as we are discussing building with tracepoints by default in the binary packages.

**Comment by christophebedard** [comment]
In this case, this tracepoint is only triggered during initialization, so the overhead is fairly small. But, yes, being able to not do this kind of processing if a tracepoint is disabled at runtime would be much better.

Since that change (as described here https://github.com/ros2/ros2_tracing/pull/43#issuecomment-1429001945) requires modifications to `tracetools` and to potential downstream packages (including `rclcpp` for this exact instance), I could first add `TRACEPOINT_ENABLED(...)` and `DO_TRACEPOINT(...)` to `tracetools`. Then we'd only need to modify `rclcpp` once.

**Review by mjcarroll — APPROVED** [review]

**Comment by christophebedard** [comment]
Testing `--packages-above tracetools test_tracetools rclcpp`:

* Linux [Build Status](http://ci.ros2.org/job/ci_linux/18089/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/12621/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/18733/)

Includes:

* https://github.com/ros2/ros2_tracing/pull/43
* This PR

**Review by mjcarroll — APPROVED** [review]
Another approval for good measure

**Comment by christophebedard** [comment]
@mjcarroll thanks :laughing:

I can't merge this PR, so I'll let you merge it.

**Pull request merged**
